scene/main/canvas_item.cpp

```diff
@@ -556,6 +556,7 @@ void CanvasItem::draw_multiline_colors(const Vector<Point2> &p_points, const Vec
 
 void CanvasItem::draw_rect(const Rect2 &p_rect, const Color &p_color, bool p_filled, real_t p_width) {
 	ERR_FAIL_COND_MSG(!drawing, "Drawing is only allowed inside NOTIFICATION_DRAW, _draw() function or 'draw' signal.");
+	ERR_FAIL_COND_MSG(p_width < 0, "The draw_rect() \"width\" argument must be non-negative.");
```

Comment:
I'm not convinced about disallowing to draw a rect with line primitives. I think it would make more sense to make the `p_width` work the same as in `draw_line` etc. and just add a relevant note in `draw_rect`'s docs about its limitations (like potential brighter corner for translucent colors caused by the overlap, missing pixels at the corners, etc.).

E.g. for debugging you often don't really care about these issues, you just want to draw a simple rect (and width not scaling with the zoom might be desired).

Also with such change now by default non-filled rects would be drawn as quads so I wonder how it would affect all built-in controls / editor plugins. I think many of them could be using `draw_rect` with default width. It could also affect performance.

So yeah, I think we should allow thin rects despite of all their quirks.

Reply:
Done.